src/controllers/employees.js

```diff
@@ -1,156 +1,60 @@
-import express from 'express';
-import fs from 'fs';
-import employees from '../models/Employees';
-
-const router = express.Router();
-
-router.post('/create', (req, res) => {
-  const employeeName = req.body.first_name;
-  const employeeLastName = req.body.last_name;
-  const employeeEmail = req.body.email;
-  const employeePassword = req.body.password;
-  const employeeTask = req.body.task;
-  const employeeProject = req.body.projects;
-  const newEmployee = {
-    id: new Date().getTime().toString().substring(6),
-    first_name: employeeName,
-    last_name: employeeLastName,
-    email: employeeEmail,
-    task: employeeTask,
-    password: employeePassword,
-    projects: employeeProject,
-  };
-  employees.push(newEmployee);
-  fs.writeFile('src/data/employees.json', JSON.stringify(employees, null, 2), (err) => {
-    if (err) {
-      res.send('Cannot create new employee');
-    } else {
-      res.send('New employee created successfully');
+import Employees from '../models/Employees';
+
+const deleteEmployees = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const result = await Employees.findByIdAndDelete(id);
+    if (!result) {
+      throw new Error(`Employee with id ${req.query.id} not found`);
     }
-  });
-});
-
-router.put('/edit', (req, res) => {
-  const employeeId = req.body.id;
-  const employeeName = req.body.first_name;
-  const employeeLastName = req.body.last_name;
-  const employeeEmail = req.body.email;
-  const employeePassword = req.body.password;
-  const employeeTask = req.body.task;
-  const employeeProject = req.body.projects;
-  const employeeToBeEdited = {
-    id: employeeId,
-    first_name: employeeName,
-    last_name: employeeLastName,
-    email: employeeEmail,
-    task: employeeTask,
-    password: employeePassword,
-    projects: employeeProject,
-  };
-  const newEmployeesList = employees.map((employee) => {
-    if (employee.id === employeeId) {
-      return employeeToBeEdited;
-    }
-    return employee;
-  });
-  fs.writeFile('src/data/employees.json', JSON.stringify(newEmployeesList, null, 2), (err) => {
-    if (err) {
-      res.send('Cannot update employee');
-    } else {
-      res.send('Employee updated successfully');
+    return res.status(200).json({
+      message: `Employee with id ${id} deleted`,
+      data: result,
+      error: false,
+    });
+  } catch (error) {
+    let status = 404;
+    if (!error.message.includes('Employee not found')) {
+      status = 404;
```

Comment:
Status is always 404 with this logic, also, the error message will never match. Please try to understand how this logic works!

```suggestion
    let status = 400;
    if (error.message.includes('Employee with id')) {
      status = 404;
```